boofuzz/sessions.py

```diff
@@ -910,20 +921,16 @@ def pre_send(self, target):
             target (session.target): Target we are sending data to
         """
 
-        if self._pre_send_methods is not None:
+        if len(self._pre_send_methods) > 0:
             try:
                 for f in self._pre_send_methods:
                     self._fuzz_data_logger.open_test_step('Pre_Send callback: "{0}"'.format(f.__name__))
                     f(target=target, fuzz_data_logger=self._fuzz_data_logger, session=self, sock=target)
             except Exception:
                 self._fuzz_data_logger.log_error(constants.ERR_CALLBACK_FUNC.format(func_name="pre_send")
                                                  + traceback.format_exc())
-                raise exception.BoofuzzCallbackError()
-            finally:
-                self._fuzz_data_logger.open_test_step('Cleaning up connections from callbacks')
                 target.close()
-        else:
-            pass
+                raise exception.BoofuzzCallbackError()
```

Comment:
Remove exception raise, just like in `_post_send`. Do the same for the restart methods section.

Reply:
Will change.